packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -222,14 +223,48 @@ const duplicateNameError = async (context: HookContext<LocationService>) => {
   }
 }
 
+/* BEFORE FIND HOOKS */
+/**
+ * @function verifyLocationParams
+ * @param context
+ * @description This hook function ensure that location find service can't be
+ * executed with empty query params object, followed by a validation that
+ * only allowed params can be included in the object.
+ */
+const verifyLocationParams = async (context: HookContext<LocationService>) => {
+  const allowed: string[] = [
+    'action',
+    'studio',
+    'slugifiedName',
+    '$limit',
+    '$sort',
+    'sceneId',
+    'projectId',
+    'name',
+    'id'
+  ]
+  const keys: string[] = Object.keys(context.params.query || {})
+  if (!keys.length) throw new BadRequest(`Query can't have empty params object`)
+  if (!keys.map((key) => allowed.includes(key)).reduce((p, c) => (!p || !c ? false : true)))
```

Comment:
schemaHooks.validateQuery(locationQueryValidator) already makes sure the query only includes parameters from locationQuerySchema and throws an error if there's anything extra.

![Screenshot 2024-09-12 at 4 50 14 p m](https://github.com/user-attachments/assets/20e73660-197b-4328-a665-fcc7d79279e2)

Reply:
@hanzlamateen I've reviewed what @jose-galvan is mentioning by analyzing what `schemaHooks.validateQuery(locationQueryValidator)` hook is doing under the hood and my what I understand is that he is right, unless we are missing something it seems that validating the parameters again if action != admin is redundant.

I introduced my implementation based off the Story conversation we had in Jira and during the Stand Up call, but I totally missed what Jose found and is mentioning. 

Regardless, since you explained the requirements I would like to know what your opinion is.

I still believe that if isAction('admin') then we should keep verifyScope('location', 'read'), there's no discussion on that one, but....

!isAction('admin') then... Do we really need to verify the query params to ensure only params being used currently are allowed? IMO it's redundant.

There still 1 more requirement to be discussed and that is Not Allowing empty query params when calling Location.find service.

So, I believe the end result should be something like

```tsx
iff(
  isProvider('external'),
  iffElse(isAction('admin'), [verifyScope('location', 'read')], [verifyEmptyQueryParams])
)
```
thoughts?

Reply:
Another thought,

A hook to validate that the query params object isn't empty is super generic, I won't be surprised if a hook making that validation already exists, if it does exist already I would appreciate if you mention which hook is, so I can simply leverage it instead of creating a new one for the same purpose.

Thanks

Reply:
So you mentioned `!isAction('admin')` in 1st comment. Its important to notice we have an overloaded scenario in MT that introduces `action: 'account'`.

https://github.com/theinfinitereality/irpro-multitenancy/blob/b1c7af5e227663c689e06f0268a9f5c27fd25ac7/services/hooks/social/location.hooks.ts#L76

We dont have an existing hook that checks for empty nor there is a documentation. Maybe you relevant one here: https://hooks-common.feathersjs.com/hooks.html

We can leverage schema validator and we should get rid of the properties that are not used. Like you mentioned (maxUsersPerInstance, isFeatured or isLobby). You can search for Type.Union and Type.Intersection in our code base to see how we can have multiple combinations.

Though its important to mention we also want to restrict some scenarios like following to be done from backend only or from admin. Here %like allows searching through api, we dont want any user to search locations without enough scopes.

https://github.com/ir-engine/ir-engine/blob/a35c1b9355c056b8b58c07a18ffe6492e5a45a71/packages/server-core/src/projects/project/project.hooks.ts#L455
https://github.com/ir-engine/ir-engine/blob/a35c1b9355c056b8b58c07a18ffe6492e5a45a71/packages/client-core/src/user/components/UserMenu/menus/LocationMenu.tsx#L80

I think `LocationMenu` is not being used anywhere so it might be redundant.

Reply:
@jonathan-casarrubias FYI